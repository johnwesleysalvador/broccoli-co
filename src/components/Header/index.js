import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Typography from '@material-ui/core/Typography';

const Wrapper = styled.div`
    min-height: 30px;
    padding: 20px;
    position: fixed;
    top: 0;
    right: 0;
    width: 100%;
    z-index: 2;
    left: 0;
    color: white;
    transition: background 300ms ease-in;

    ${({ isOnScroll }) =>
        isOnScroll &&
        `
        color: initial;
        background: white;
    `};
`;

class Header extends PureComponent {
    state = {
        isOnScroll: false,
    };
    componentDidMount() {
        window.addEventListener('scroll', this.handleScroll);
    }

    handleScroll = () => {
        const { pageYOffset } = window;

        if (pageYOffset > 0) {
            this.setState({ isOnScroll: true });
        } else {
            this.setState({ isOnScroll: false });
        }
    };
    render() {
        const { children } = this.props;
        return (
            <Wrapper isOnScroll={this.state.isOnScroll}>
                <Typography variant="title" color="inherit">
                    {children}
                </Typography>
            </Wrapper>
        );
    }
}

Header.propTypes = {
    children: PropTypes.string.isRequired,
};

export default Header;
