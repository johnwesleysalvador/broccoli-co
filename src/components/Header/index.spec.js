import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Header from '.';

describe('<Header />', () => {
    test('Should render Header', () => {
        const render = renderer.create(<Header>Lorem Ipsum</Header>).toJSON();
        expect(render).toMatchSnapshot();
    });
});
