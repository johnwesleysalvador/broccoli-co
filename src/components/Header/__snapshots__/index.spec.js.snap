// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render Header 1`] = `
.c0 {
  min-height: 30px;
  padding: 20px;
  position: fixed;
  top: 0;
  right: 0;
  width: 100%;
  z-index: 2;
  left: 0;
  color: white;
  -webkit-transition: background 300ms ease-in;
  transition: background 300ms ease-in;
}

<div
  className="c0"
>
  <h2
    className="MuiTypography-root-1 MuiTypography-title-7 MuiTypography-colorInherit-20"
  >
    Lorem Ipsum
  </h2>
</div>
`;
